web3.js/src/connection.ts

```diff
@@ -767,6 +767,8 @@ export type InflationReward = {
   amount: number;
   /** post balance of the account in lamports */
   postBalance: number;
+  /** vote account commission when the reward was credited */
+  commission: number | null;
```

Comment:
Are you sure this is right? The [JSON-RPC docs](https://docs.solana.com/developing/clients/jsonrpc-api#results-20) describe this property as `commission: <u8|undefined>`.

```suggestion
  commission?: number;
```

Reply:
You're correct! Will fix.

Reply:
I mean, I'm _really_ just asking. It's a distinct possibility that the docs are wrong. https://github.com/solana-labs/solana/pull/27655

Reply:
So it is returning null from the request, when doing the live testing it does seem to require that the field is nullable. And I know from my own use that making the request to mainnet the JSON response is returning the commission (at least for InflationReward) as a number.

But I'm fixing right now which I suppose supporting the `optional(nullable(number()))` more closely matches the JSON RPC spec with it being ABLE to be undefined (although I'm not certain it does return without the field).

Reply:
Allowing for `undefined` would definitely protect against a response from an old RPC fataling the client. I don't think there are any RPCs old enough to return `undefined` for this property but ¯\_(°ペ)_/¯

Reply:
Indeed and agreed. Updated.